### PR TITLE
TRUNK-6202 Replace Hibernate Criteria API with JPA for HibernateOrderDAO

### PR DESCRIPTION
**Pull request by k4pran**
## Description of what I changed

This task is to migrate the deprecated hibernate criteria to jpa criteria for the HibernateOrderDAO class.

## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-6202

## Checklist: I completed these to help reviewers :)

- [X] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [X] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [X] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [X] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [X] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api/src/main/java/org/openmrs/api/db/hibernate/HibernateOrderDAO.java [hunk 1]
```diff
@@ -692,53 +777,81 @@ public void purgeOrderType(OrderType orderType) {
...
-		Criteria criteria = sessionFactory.getCurrentSession().createCriteria(OrderType.class);
-		criteria.add(Restrictions.eq("parent", orderType));
+		Session session = sessionFactory.getCurrentSession();
+		CriteriaBuilder cb = session.getCriteriaBuilder();
+		CriteriaQuery<OrderType> cq = cb.createQuery(OrderType.class);
+		Root<OrderType> root = cq.from(OrderType.class);
+
 		if (!includeRetired) {
-			criteria.add(Restrictions.eq("retired", false));
+			cq.where(cb.equal(root.get("parent"), orderType), cb.isFalse(root.get("retired")));
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/api/db/hibernate/HibernateOrderDAO.java:786]: Shouldn't `cq.where(cb.equal(root.get("parent")` be outside the if else clause?

File: api/src/test/java/org/openmrs/api/OrderServiceTest.java [hunk 2]
```diff
@@ -4101,4 +4137,19 @@ public void getOrderAttributeByUuid_shouldReturnOrderAttributeUsingProvidedUuid(
 		assertEquals("Testing Reference", orderAttribute.getValueReference());
 		assertEquals(1, orderAttribute.getId());
 	}
+
+	@Test
+	public void getOrderAttributeTypeByName_shouldReturnCorrectOrderAttributeType() {
+		String orderAttributeTypeName = "Supplies";
```
Comment (dkayiwa) [inline api/src/test/java/org/openmrs/api/OrderServiceTest.java:4143]: https://wiki.openmrs.org/display/docs/Java+Conventions#JavaConventions-NamingConventions

File: api/src/main/java/org/openmrs/api/db/hibernate/HibernateOrderDAO.java (hunk 1 above)
Comment (k4pran) [reply api/src/main/java/org/openmrs/api/db/hibernate/HibernateOrderDAO.java]: Moved to outside

File: api/src/test/java/org/openmrs/api/OrderServiceTest.java (hunk 2 above)
Comment (k4pran) [reply api/src/test/java/org/openmrs/api/OrderServiceTest.java]: Added final and capitalized

**Pull request merged**